### PR TITLE
feat: add test workflow

### PR DESCRIPTION
**Pull request by itegulov**
ditto

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v2
+
+      - name: Check Formatting
+        run: cargo fmt -p near-abi-client -p near-abi-client-macros -p near-abi-client-impl -- --check
+
+      - name: Check Clippy
+        run: cargo clippy -p near-abi-client -p near-abi-client-macros -p near-abi-client-impl --tests -- -Dclippy::all
```
Comment (miraclx) [inline .github/workflows/test.yml:62]: Are the explicit package definitions useful here?

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Build files
-/target
+**/target
```
Comment (miraclx) [inline .gitignore]: I believe just `target` should be fine. Or `target/` if you want to be directory specific.

File: near-abi-client/Cargo.toml [hunk 3]
```diff
@@ -2,6 +2,7 @@
 name = "near-abi-client"
 version = "0.1.0"
 edition = "2021"
+rust-version = "1.56.0"
```
Comment (miraclx) [inline near-abi-client/Cargo.toml:5]: 👍🏽

**Review by miraclx — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (miraclx) [reply .github/workflows/test.yml:62]: Just saw https://github.com/near/near-sdk-abi/pull/5#discussion_r1060295789, got it.

File: .gitignore (hunk 2 above)
Comment (itegulov) [reply .gitignore]: Ah, I was confused because I had some residual `examples/*/target` dirs back from when they used to be standalone.

Comment (miraclx) [reply .gitignore]: Yeah, considering we don't have those anymore, it's unclear if we should be specific and ignore just the root target _folder_ (`/target/`) or a target folder at any depth (`target/` or `**/target/`) or any file (dirs inclusive) named target (`target`).

Do you prefer the opt-out approach of starting strict and revising the specifier when we have an actual need for a file/folder named target, which is unlikely.

Or ignore specifically for our case and selectively revise the specifier to include more and more cases as we discover them.

I'm leaning towards the first. But to be folder specific – `target/`.

Comment (itegulov) [reply .gitignore]: I don't have a strong opinion on this, so let's just go with `**/target/` as you have suggested

**Pull request merged**